plugins/inputs/t128_graphql/t128_graphql.go

```diff
@@ -0,0 +1,284 @@
+package t128_graphql
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Jeffail/gabs"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const (
+	//make this configurable?
+	DefaultRequestTimeout = time.Second * 5
+)
+
+type T128GraphQL struct {
+	CollectorName string            `toml:"collector_name"`
+	BaseURL       string            `toml:"base_url"`
+	EntryPoint    string            `toml:"entry_point"`
+	Fields        map[string]string `toml:"extract_fields"`
+	Tags          map[string]string `toml:"extract_tags"`
+
+	Client *http.Client
+	Query  string
+}
+
+var sampleConfig = `
+## Collect data using graphQL
+[[inputs.t128_graphql]]
+## Required. The telegraf collector name
+# collector_name = "arp-state"
+
+## Required. GraphQL ports vary across 128T versions
+# base_url = "http://localhost:31517"
+
+## The starting point in the graphQL tree for all configured tags and fields
```

Comment:
@salegria8 how do we configure the actual query in this sample config?

Reply:
You got it right a few comments down - I use the configured entryPoint, fields and tags to create the query. Next step is to test with other queries. There are some kinds of queries I need to handle carefully but they came up in other comments so I'll add more detail there.